pilot/pkg/networking/core/v1alpha3/tls.go

```diff
@@ -308,7 +310,8 @@ TcpLoop:
 		}
 		out = append(out, &filterChainOpts{
 			destinationCIDRs: []string{destinationCIDR},
-			networkFilters:   buildOutboundNetworkFiltersWithSingleDestination(push, node, statPrefix, clusterName, "", listenPort, destinationRule),
+			networkFilters: buildOutboundNetworkFiltersWithSingleDestination(push, node, statPrefix, clusterName, "",
+				listenPort, destinationRule, tunnelingconfig.NilBuilder),
```

Comment:
Same question - who not ?

Reply:
In this case, network filters are created for ports without associated virtual services, so it's the same case as `AUTO_PASSTHROUGH` explained above. Another problem is that the destination rule used to apply to this filter is not a destination rule for some destination host, but a destination rule for this particular service.

Applying a destination rule for a port with associated virtual service:
```
service := push.ServiceForHostname(node, host.Name(routes[0].Destination.Host))
destinationRule = CastDestinationRule(node.SidecarScope.DestinationRule(
        model.TrafficDirectionOutbound, node, service.Hostname))
```

Reply:
I don't get it. What does virtual service have to do with DestinationRule? They are orthogonal?

Reply:
Ah read the comment elsewhere. IMO just because users would _typically_ use tunnelling with a VirtualService doesn't mean we should invalidate the current API model. If they want to use tunnel directly, why not...

Reply:
> Another problem is that the destination rule used to apply to this filter is not a destination rule for some destination host, but a destination rule for this particular service.

Please ignore this argument, it doesn't make sense. I misunderstood the implementation, sorry.

> IMO just because users would typically use tunnelling with a VirtualService doesn't mean we should invalidate the current API model.

Yes, you're right. Listeners without associated virtual services should also apply tunnel settings. I will do that.

Reply:
@howardjohn I thought once again about this case and it can't work for TLS traffic, because it will cause the same problem as in case of AUTO_PASSTHROUGH gateway - I mean TLS handshake error. The only working case might be plain traffic, but I really doubt that it might be useful and I guess that it would bring only troubles and confusion, not a benefit. If TLS traffic can't be tunneled in these cases, then I think we shouldn't support TCP as well.

Reply:
Sorry, could you expand more on how TCP works but TLS doesn't? They are essentially same - an opaque stream of bytes?

Reply:
Ok, so let's say I want to tunnel traffic to the `target.com` service via `proxy.com` service and there is no associated virtual service with `target.com`. Without virtual service, the only way to enable tunneling to the `target.com` is to apply the following resource:
```
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: tunnel
spec:
  host: proxy.com
  trafficPolicy:
    tunnel:
      protocol: connect
      targetHost: target.com
      targetPort: 443
```
Then the connection would work as follows:
1. A client send a request to: `https://proxy.com:3128`.
2. The request goes to TcpProxy filter which applies tunneling and sends `HTTP CONNECT target.com:443`.
3. Once a connection is established, istio-proxy sends data from the downstream: it is `TLSv1.2 Client Hello SNI=proxy.com`.
4. The `target.com` server returns `TLSv1.2 Server Hello` and returns its certificate details `subject: CN=target.com`.

Now this TLS connection can be recognized as insecure and be rejected by the client, because requested server name and the subject or SAN don't match. This handshake might succeed if client agrees for that, e.g. `curl --insecure`, but I doubt that someone would like to do that. In my opinion it's tricky, insecure and non-intuitive.

Reply:
As requested in the comments below, I added applying of tunneling_config to listeners without virtual services, except AUTO_PASSTHROUGH.